src/Psy/Command/ExternalEditorCommand.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Psy\Command;
+
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ExternalEditorCommand extends Command
+{
+    protected function configure()
+    {
+        $this
+            ->setName('externaledit')
+            ->setAliases(array('edit'))
+            ->setDefinition(array(
+                new InputArgument('file', InputArgument::OPTIONAL, 'The file to open for editing. If this is not given, edits a temporary file.', null),
+            ))
+            ->setDescription('Open an external editor. Afterwards, get produced code in input buffer.')
+            ->setHelp('Set the EDITOR environment variable to something you\'d like to use.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $filePath = $input->getArgument('file');
+
+        if ($filePath === null) {
+            $filePath = tempnam(sys_get_temp_dir(), 'psysh');
```

Comment:
Please pass in the "runtime dir" from Shell::config in the constructor and use that rather than `sys_get_temp_dir`. It's effectively the same, but can be overridden via XDG for systems that support it.

Reply:
Also, please name the tempfile with `externaledit` or similar as the prefix; `psysh` is too generic.